### PR TITLE
[StaticRegistrar] Switch to dynamically linking the registrar

### PR DESCRIPTION
**Pull request by Therzok**
Fixes VSTS #568281 VS for Mac Fails to start with 7.5.0.421(7ba471f)

### CODE REVIEW

**Comment by Therzok** [comment]
what did I do.

**Comment by Therzok** [comment]
Currently marked as do not merge. Need a xammac build of 4.3.x that can be used for the new mmp like arguments.

**Comment by Therzok** [comment]
This also fixes some other issues:
https://github.com/mono/monodevelop/issues/3924
The need to have a newer XamMac to _run_ VSM (it works under 4.1.x as well as 4.3.x)

File: main/build/MacOSX/monostub.mm [hunk 1]
```diff
@@ -413,7 +391,31 @@ int main (int argc, char **argv)
...
+			fprintf (stderr, "Failed to load libxammac.dylib: %s\n", dlerror ());
+			NSString *msg = @"This application requires Xamarin.Mac native library side-by-side.";
+			exit_with_message ((char *)[msg UTF8String], argv[0]);
+		}
+	}
+
+#if STATIC_REGISTRAR
+	libvsmregistrar = dlopen ("@loader_path/vsmregistrar.dylib", RTLD_LAZY);
+	if (!libvsmregistrar) {
+		libvsmregistrar = dlopen ("@loader_path/../Resources/lib/monodevelop/bin/libvsmregistrar.dylib", RTLD_LAZY);
```
Comment (alanmcgovern) [inline main/build/MacOSX/monostub.mm]: `vsmregistrar` or `libvsmregistrar` ? Can we choose one and use it consistently?

Comment (Therzok) [reply main/build/MacOSX/monostub.mm]: Oh god, we need libvsmregistrar, good catch.

File: version-checks [hunk 2]
```diff
@@ -17,8 +17,8 @@ DEP[0]=md-addins
 DEP_NAME[0]=MDADDINS
 DEP_PATH[0]=${top_srcdir}/../md-addins
 DEP_MODULE[0]=git@github.com:xamarin/md-addins.git
-DEP_NEEDED_VERSION[0]=940b03fe3ff53201ce1e19b241671be5483104bd
-DEP_BRANCH_AND_REMOTE[0]="master origin/master"
+DEP_NEEDED_VERSION[0]=6395b223fc37acdef4ad2ba8bfe9944f63212063
+DEP_BRANCH_AND_REMOTE[0]="static-registrar-bundle-path origin/static-registrar-bundle-path"
```
Comment (alanmcgovern) [inline version-checks]: oops?

**Pull request merged**